`init` tox tests failing on macos 11

### Bug Description

On macos 11 (but not on Linux, Windows or macos 12), the tox tests after creating a charm fail. The failures appear to be about the path not being correctly set or something.

### To Reproduce

On macos 11:
1. `charmcraft init`
2. `tox -v`

See: https://github.com/canonical/charmcraft/actions/runs/7788640169/job/21238497566?pr=1507

### Environment

macos 11

### charmcraft.yaml

```shell
n/a
```


### Relevant log output

```shell
___________________________ test_tox_success[simple] ___________________________
  new_path = PosixPath('/Users/runner/work/charmcraft/charmcraft/.tox/tox_tmp/test-py3.10/pytest-of-runner/pytest-0/test_tox_success_simple_0')
  init_command = <charmcraft.application.commands.init.InitCommand object at 0x10b4c4520>
  profile = 'simple'
      @pytest.mark.slow()
      @pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
      @pytest.mark.skipif(bool(os.getenv("RUNNING_TOX")), reason="does not work inside tox")
      @pytest.mark.parametrize("profile", list(commands.init.PROFILES))
      def test_tox_success(new_path, init_command, profile):
          # fix the PYTHONPATH and PATH so the tests in the initted environment use our own
          # virtualenv libs and bins (if any), as they need them, but we're not creating a
          # venv for the local tests (note that for CI doesn't use a venv)
          env = os.environ.copy()
          env_paths = [p for p in sys.path if "env/lib/python" in p]
          if env_paths:
              if "PYTHONPATH" in env:
                  env["PYTHONPATH"] += ":" + ":".join(env_paths)
              else:
                  env["PYTHONPATH"] = ":".join(env_paths)
              for path in env_paths:
                  bin_path = path[: path.index("env/lib/python")] + "env/bin"
                  env["PATH"] = bin_path + ":" + env["PATH"]
      
          init_command.run(create_namespace(profile=profile))
      
          result = subprocess.run(
              ["tox", "-v"],
              cwd=new_path,
              env=env,
              text=True,
              stdout=subprocess.PIPE,
              stderr=subprocess.STDOUT,
              check=False,
          )
          if result.returncode != 0:
  >           raise AssertionError(
                  "".join(("Tox run failed:\n", *(f":: {line}" for line in result.stdout.splitlines())))
              )
  E           AssertionError: Tox run failed:
  E           :: format: find interpreter for spec PythonSpec(path=/Users/runner/work/charmcraft/charmcraft/.tox/.tox/bin/python):: format: proposed PythonInfo(spec=CPython3.12.1.final.0-64, system=/Library/Frameworks/Python.framework/Versions/3.12/bin/python3.12, exe=/Users/runner/work/charmcraft/charmcraft/.tox/.tox/bin/python, platform=darwin, version='3.12.1 (v3.12.1:2305ca5144, Dec  7 2023, 17:23:39) [Clang 13.0.0 (clang-1300.0.29.30)]', encoding_fs_io=utf-8-utf-8):: format: create virtual environment via CPython3Posix(dest=/Users/runner/work/charmcraft/charmcraft/.tox/format, clear=False, no_vcs_ignore=False, global=False):: format: add seed packages via FromAppData(download=False, pip=bundle, via=copy, app_data_dir=/Users/runner/Library/Application Support/virtualenv):: format: add activators for Bash, CShell, Fish, Nushell, PowerShell, Python:: format: install_deps> python -I -m pip install ruff:: format: exit 0 (3.08 seconds) /Users/runner/work/charmcraft/charmcraft/.tox/tox_tmp/test-py3.10/pytest-of-r
  /Users/runner/work/charmcraft/charmcraft/tests/integration/commands/test_init.py:253: AssertionError
```

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -47,7 +47,7 @@ jobs:
   run-tests:
     strategy:
       matrix:
-        os: [ubuntu-22.04, macos-11, macos-12, windows-2019, windows-2022]
+        os: [ubuntu-22.04, macos-12, macos-13, windows-2019, windows-2022]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout code

--- a/tests/integration/commands/test_init.py
+++ b/tests/integration/commands/test_init.py
@@ -18,7 +18,6 @@ import argparse
 import contextlib
 import os
 import pathlib
-import platform
 import re
 import shutil
 import subprocess
@@ -222,10 +221,6 @@ def test_executable_set(new_path, init_command):
 
 @pytest.mark.slow()
 @pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
-@pytest.mark.xfail(
-    platform.system() == "Darwin" and platform.platform().startswith("macOS-11."),
-    reason="Currently failing on macos 11, see https://github.com/canonical/charmcraft/issues/1508",
-)
 @pytest.mark.skipif(bool(os.getenv("RUNNING_TOX")), reason="does not work inside tox")
 @pytest.mark.parametrize("profile", list(commands.init.PROFILES))
 def test_tox_success(new_path, init_command, profile):

